test/Governance.test.ts

```diff
@@ -169,30 +174,129 @@ describe('Governance', function () {
         proposalCalldatas,
         descriptionHash
       )
-      await this.governance.connect(this.voter).castVote(proposalId, 1)
-      await this.governance.connect(this.delegatee1).castVote(proposalId, 1)
-      await this.governance.connect(this.delegatee2).castVote(proposalId, 1)
-      await this.governance.connect(this.delegatee3).castVote(proposalId, 1)
-      await this.governance.connect(this.newDelegatee).castVote(proposalId, 1)
+      expect(await this.governance.state(proposalId)).to.equal(0)
+
+      this.castVoteAndCheck = async function (
+        proposalId: string,
+        proposalRoles: string[],
+        voter: SignerWithAddress,
+        role: string
+      ): Promise<void> {
+        await this.governance.connect(voter).castVote(proposalId, 1)
+        const receipt = await this.governance.getReceipt(
+          proposalId,
+          voter.address
+        )
+        if (proposalRoles.length === 0) {
+          expect(receipt.votes).to.equal(this.baseVotingPower)
+          expect(receipt.support).to.equal(1)
+        } else {
+          let hasRole = false
+          for (const proposalRole of proposalRoles) {
+            if (proposalRole === role) {
+              expect(receipt.votes).to.equal(this.roleVotingPower)
+              expect(receipt.support).to.equal(1)
+              hasRole = true
+            }
+          }
+          if (!hasRole) {
+            expect(receipt.votes).to.equal(0)
+            expect(receipt.support).to.equal(1)
+          }
+        }
+      }
+
+      await this.castVoteAndCheck(proposalId, proposalRoles, this.voter, '')
+      await this.castVoteAndCheck(
+        proposalId,
+        proposalRoles,
+        this.delegatee1,
+        DEVELOPER_ROLE
+      )
+      await this.castVoteAndCheck(
+        proposalId,
+        proposalRoles,
+        this.delegatee2,
+        LEGAL_ROLE
+      )
+      await this.castVoteAndCheck(
+        proposalId,
+        proposalRoles,
+        this.delegatee3,
+        TREASURY_ROLE
+      )
+      await this.castVoteAndCheck(
+        proposalId,
+        proposalRoles,
+        this.newDelegatee,
+        ''
+      )
+
       await advanceBlockTo((await provider.getBlockNumber()) + PROPOSAL_SPAN)
+      expect(await this.governance.state(proposalId)).to.equal(4)
       await this.governance.queue(
         proposalTargets,
         proposalValues,
         proposalCalldatas,
         descriptionHash
       )
+      expect(await this.governance.state(proposalId)).to.equal(5)
       await this.governance.execute(
         proposalTargets,
         proposalValues,
         proposalCalldatas,
         descriptionHash
       )
-      const state = await this.governance.state(proposalId)
-      expect(state).to.equal(7)
+      expect(await this.governance.state(proposalId)).to.equal(7)
       return proposalId
     }
   })
 
+  it('Verify initial governance state', async function () {
```

Comment:
praise: the verifying of initial state is exactly the kind of thing I was referring to in the previous round of comments 👍